src/entity/posts/utils.ts

```diff
@@ -248,13 +248,14 @@ export const generateTitleHtml = (
 ): string =>
   `<p>${renderMentions(markdown.utils.escapeHtml(title), mentions)}</p>`;
 
-export const createSharePost = async (
+export const upsertSharePost = async (
```

Comment:
I went ahead with re-using the existing mutation rather than creating a new one. Let me know your thoughts

Reply:
I agree with Chris it should be separate thing.